ddev/src/ddev/repo/config.py

```diff
@@ -0,0 +1,29 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from functools import cached_property
+
+
+class RepositoryConfig:
+    def __init__(self, data: dict):
+        self.__data = data
+
+    @property
+    def data(self) -> dict:
```

Comment:
Since we're exposing the only attribute that the instance has anyway, what do you think about making `RepositoryConfig` a subclass of [`UserDict`](https://docs.python.org/3/library/collections.html#collections.UserDict)?

Reply:
I don't think we should treat the class like a `dict`. Also, iirc that's a legacy thing that was begrudgingly kept in py3 https://docs.python.org/3/library/collections.html#userdict-objects:

> The need for this class has been partially supplanted by the ability to subclass directly from dict